### PR TITLE
refactor: enable perl signature `Text.pm`

### PR DESCRIPTION
**Pull request by dipietroR**
Replacing "shift" prototypes with perl signatures

### CODE REVIEW

**Review by stephanegigandet — APPROVED** [review]
Thanks!

**Comment by alexgarel** [comment]
I update with rebase to see if tests would pass (error seems in an unrelated test)

**Comment by alexgarel** [comment]
For tests, the error is the following

```
[Fri Aug 19 14:47:41.338524 2022] [:error] [pid 29] Too few arguments for subroutine 'ProductOpener::Text::remove_tags_and_quote' at /opt/product-opener/lib/ProductOpener/Users.pm line 311.\n
```

to see it you have to run `make tail` in a first window, then run

```
make test-int test=create_user.t
```

the log will be one of the last.

**Comment by alexgarel** [comment]
another error:
```
[Fri Aug 19 14:57:18.403300 2022] [:error] [pid 32] [Fri Aug 19 14:57:18 2022] -e: Too few arguments for subroutine 'ProductOpener::Text::remove_tags_and_quote' at /opt/product-opener/lib/ProductOpener/Food.pm line 2616.\n
```

**Comment by dipietroR** [comment]
@stephanegigandet 35a8c00dfeba3b927f3ecfa12f8db521e14782b8 and 4a0e787d776387cd4a1604ad495bf1d0fe1f5b6f don't appear to be related to this PR and it seems to be causing some tests to fail (specifically Check perl), if they are related could you please explain how?

**Comment by dipietroR** [comment]
I tried to update this branch but accidentally got rid of @stephanegigandet 's commits (sorry!!).

I'm not sure if they were relevant to this PR, if they are could you add it back and explain how?

**Comment by Ban3** [comment]
Failures were in the integration tests for creating a user and a product. Didn't seem related to this PR.

I restarted the job in case it was just something temporary.

**Comment by Ban3** [comment]
Looks like still failing. Not sure what's going on there.

https://github.com/openfoodfacts/openfoodfacts-server/runs/8001604686?check_suite_focus=true#step:5:12555

**Comment by alexgarel** [comment]
@Ban3 @yuktea  and @stephanegigandet this lack of visibility of errors while running integration tests made me think we could do this https://github.com/openfoodfacts/openfoodfacts-server/issues/7260

File: cgi/user.pl [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/perl -w
+remove_tags_and_quote(param#!/usr/bin/perl -w
```
Comment (Ban3) [inline cgi/user.pl:1]:
```suggestion
#!/usr/bin/perl -w
```

Comment (dipietroR) [reply cgi/user.pl:1]: Thank you!

**Review by stephanegigandet — APPROVED** [review]
Thank you!

**Pull request merged**